### ISSUE TITLE
ProjectGenerator add-on libs  pathing issue and src subfolders not linked correctly 

### ISSUE THREAD

**Issue by danoli3** [openframeworks/projectGenerator#515]
For Addon libs:
<img width="1245" alt="Screenshot 2024-06-05 at 2 08 09 am" src="https://github.com/openframeworks/projectGenerator/assets/830748/9eed664d-6dd1-4563-b43c-55027032f87b">

Where it should look like this:
<img width="800" alt="Screenshot 2024-06-05 at 3 10 35 am" src="https://github.com/openframeworks/projectGenerator/assets/830748/a3d82e6e-b869-4dd2-983e-f995ab98054b">

had a Quick Look looks related to the paths not being correct 

```
                    if (folder.begin()->string() == "addons") {
                        addCommand("Add :objects:"+thisUUID+":sourceTree string <group>");
                        fs::path addonFolder { fs::path(fullPath).filename() };
                        addCommand("Add :objects:"+thisUUID+":path string " + ofPathToString(addonFolder));
                        // alert ("group " + folder.string() + " : " + base.string() + " : " + addonFolder.string(), 32);
                    } else {
                        addCommand("Add :objects:"+thisUUID+":sourceTree string SOURCE_ROOT");
                    }


```

I re-added this to xcframeworks and now it works 

Also if you run the projectGenerator against the projectGenerator (inception)
It fails to link or add the other src folders in the same directory 

<img width="336" alt="Screenshot 2024-06-05 at 11 22 45 am" src="https://github.com/openframeworks/projectGenerator/assets/830748/eef55064-50cc-4957-bdec-baf8722e8723">

**Comment by danoli3** [comment]
another issue if ofApp.cpp is modified to ofApp.mm or main.cpp is modified to main.mm it doesn't update ref to those objective-c++

**Comment by danoli3** [comment]
seems like issue happening too with frameworks.... maybe some issues after code re-write @dimitre XCode more changes : PG v40 (#504)

This is just for addons

**Comment by danoli3** [comment]
Do you think this is a UUID issue?

What about this works vs the prior code:
```
addCommand("Add :objects:"+thisUUID+":sourceTree string <group>");
                        fs::path addonFolder { fs::path(fullPath).filename() };
                        addCommand("Add :objects:"+thisUUID+":path string " + ofPathToString(addonFolder));
                        ```
                        
                        something to do with group maybe

**Comment by dimitre** [comment]
I'll be testing this tomorrow.
are you having addon issues with regular or bleeding? any specific addon for testing here?

**Comment by danoli3** [comment]
any addons libs/frameworks or any source folders in the source directory of current project or addons 
Platform macOS

**Comment by dimitre** [comment]
I've updated with some changes, it is v44 now. can you please test?
it is building all addons example ok here.
pg regeneration folders look ok

**Comment by dimitre** [comment]
have a second installation here to test bleeding libs. it misses fmod, I think the folder ```libs/fmod/lib``` is empty

**Comment by danoli3** [comment]
<img width="293" alt="Screenshot 2024-06-05 at 3 04 28 pm" src="https://github.com/openframeworks/projectGenerator/assets/830748/594da824-e85c-40ad-ad6b-ab9a8934aa53">
subfolders working! 

addons still linking weird again:
<img width="714" alt="Screenshot 2024-06-05 at 3 05 53 pm" src="https://github.com/openframeworks/projectGenerator/assets/830748/c77e1bea-f7f5-4e31-a6ee-d00bc3da6587">
Here I added svgtiny manually to fix linking issues and you can see the difference, required shows up

system frameworks weird full path: 
<img width="475" alt="Screenshot 2024-06-05 at 3 05 14 pm" src="https://github.com/openframeworks/projectGenerator/assets/830748/b3837548-c9a1-4dcd-8560-5429a7768687">

**Comment by danoli3** [comment]
@dimitre it might just need realpaths instead of ../../.. which might be breaking it

**Comment by dimitre** [comment]
You are probably using an alternative template than the OF master one.
here it includes xcframeworks correctly, don't compile because it is missing the .a files.
<img width="1512" alt="Screenshot 2024-06-05 at 10 14 36" src="https://github.com/openframeworks/projectGenerator/assets/58289/59ca859a-c295-4118-9d09-fde3873dcee7">

<img width="1512" alt="Screenshot 2024-06-05 at 10 17 57" src="https://github.com/openframeworks/projectGenerator/assets/58289/697eb26c-079b-43fb-85fa-8b8d42b4e24b">

**Comment by danoli3** [comment]
Yeah same problem with addons looks like though, basically something wrong in the updated paths for addons, either uuid issue or something about the ../../../ relative thing I feel it might be

**Comment by danoli3** [comment]
![image](https://github.com/openframeworks/projectGenerator/assets/830748/434094e4-0279-4b8b-b621-596251dbcaa7)

See its missing the required which might mean its missing the link correctly, or maybe thats something to add to pg grouping commands

**Comment by dimitre** [comment]
I didn't understand your last post. what is the required?

**Comment by dimitre** [comment]
For bleeding libs work it is necessary to alter things in OF Core. 
I think we should do it in a separate branch or something to be able to use the classic libs
https://github.com/openframeworks/openFrameworks/blob/d72da64356aa46fa0ba71ec7d0302a7ddc6af38b/libs/openFrameworksCompiled/project/osx/CoreOF.xcconfig#L29-L41

**Comment by danoli3** [comment]
check https://github.com/danoli3/openFrameworks/tree/macOS-xcframeworks
should be ready to go man 

yeah its super weird the other libraries currently bind but not svgtiny, I checked the apothecary lib and everything looks good with symbols so added manually but the other 3 worked

**Comment by dimitre** [comment]
@danoli3 v45 merged. fixes system frameworks in iOS.
let me know how can I test and debug this issue

**Comment by danoli3** [comment]
Yeah still issue for xcframework for addonsExample:
This is what PG makes atm:
```
"3E4488B0-56D0-472C-B8FA-1AB8716248B5": {
			"fileEncoding": "4",
			"isa": "PBXFileReference",
			"lastKnownFileType": "folder",
			"path": "../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework",
			"sourceTree": "SOURCE_ROOT"
		},
```
		

Where manually added:
```
		BF879F872C118EB400200951 /* svgtiny.xcframework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.xcframework; name = svgtiny.xcframework; path = ../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework; sourceTree = "<group>"; };
```

And the build phase changes to:
```
/* Begin PBXFrameworksBuildPhase section */
		E4B69B590A3A1756003C02F2 /* Frameworks */ = {
			isa = PBXFrameworksBuildPhase;
			buildActionMask = 2147483647;
			files = (
				BF879F7D2C118D6900200951 /* ../../../addons/ofxAssimpModelLoader/libs/assimp/lib/osx/assimp.xcframework in Frameworks */,
				BF879F7E2C118D6900200951 /* ../../../addons/ofxOpenCv/libs/opencv/lib/osx/opencv.xcframework in Frameworks */,
				BF879F882C118EB400200951 /* svgtiny.xcframework in Frameworks */,
				BF879F7F2C118D6900200951 /* ../../../addons/ofxSvg/libs/libxml2/lib/osx/libxml2.xcframework in Frameworks */,
			);
			runOnlyForDeploymentPostprocessing = 0;
		};
/* End PBXFrameworksBuildPhase section */
/* Begin PBXGroup section */
		"007FA649-C8DE-4E68-ACE2-CD1062614ADF" /* osx */ = {
			isa = PBXGroup;
			children = (
				"3E4488B0-56D0-472C-B8FA-1AB8716248B5" /* ../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework */,
			);
			name = osx;
			sourceTree = SOURCE_ROOT;
		};
```
Looks like key is:
lastKnownFileType = wrapper.xcframework

**Comment by dimitre** [comment]
I've tested now your branch to try and see the same errors as you but I can't build openFrameworks lib.
it misses utf8 and glm

**Comment by danoli3** [comment]
oh weird yeah its not in the zips, investigating

**Comment by danoli3** [comment]
Okay yep wasn't copying over to the final location for those not packaged into xcframeworks. Fixed with https://github.com/openframeworks/apothecary/pull/385

**Comment by dimitre** [comment]
Nice. can you please bring latest scripts to your macOS-xcframeworks branch @danoli3 ?

**Comment by danoli3** [comment]
Updated merged with master changes, still waiting on libraries to build, will get that cache system working

**Comment by dimitre** [comment]
Almost there. now it is complaining about libfmod
path is wrong in lib installation, two "osx" folders
```
libs/fmod/lib/osx/osx/libfmod.dylib
```

**Comment by dimitre** [comment]
I've moved libfmod to the right place and now allAddonsExample starts complaining other things
<img width="1198" alt="Screenshot 2024-06-06 at 18 48 11" src="https://github.com/openframeworks/projectGenerator/assets/58289/a421a67a-0c5d-4124-9e8b-5812a97ca48d">
<img width="1141" alt="Screenshot 2024-06-06 at 18 48 18" src="https://github.com/openframeworks/projectGenerator/assets/58289/14542224-8e36-465e-8b94-4661ef3425f2">

**Comment by dimitre** [comment]
removing ofxKinect ofxAssimpModelLoader and ofxOpenCv it runs great here.

**Comment by danoli3** [comment]
So thats the symptom of the xcframeworks not being added correctly for the addons, need that xcframework group tag it seems

**Comment by dimitre** [comment]
are you getting this errors too? 
if yes please send me instructions of how manually add them to complete allAddonsExample build.
which version of PG are you using?
<img width="793" alt="Screenshot 2024-06-07 at 01 00 41" src="https://github.com/openframeworks/projectGenerator/assets/58289/e771be4a-84cc-4cd1-90f5-7b1ad1c7faee">

**Comment by danoli3** [comment]
Oh just remove them and manually add via add to project in Xcode directly!
Then compare the project file xml for that difference

**Comment by danoli3** [comment]
Oh yep will fix the fmod lib folder depth next build! Apothecary caching is
working so well!! 40-50 mins down to seconds if no changes to target libs

**Comment by dimitre** [comment]
removed and re added. same undefined symbols error
Test yourself when you can
Greattt onee about apothecary!!

**Comment by dimitre** [comment]
Hello @danoli3 I'll wait until we can isolate the xcframeworks issues and make the needed modifications. I've seen in code some special treatment for ".framework" extension that maybe should be applied to ".xcframework" too.

**Comment by danoli3** [comment]
Working:
``
"BFE7C3462C13750500EE5EDA": {
    "isa": "PBXFileReference",
    "lastKnownFileType": "wrapper.xcframework",
    "name": "svgtiny.xcframework",
    "path": "../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework",
    "sourceTree": "<group>"
},
```

Not working:
```
"044FEE66-08B1-4F5B-B02E-B8C3B28DEF06": {
			"fileEncoding": "4",
			"isa": "PBXFileReference",
			"lastKnownFileType": "folder",
			"path": "../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework",
			"sourceTree": "SOURCE_ROOT"
		},
```

Must be just code related to group setting and lastKnownFileType to fix it

**Comment by danoli3** [comment]
```
"BEB93389-8DF0-48A5-BF6D-5321E62624B6" /* libs */ = {
			isa = PBXGroup;
			children = (
				"99E2C68C-B720-49E4-AB17-7F9847A099F4" /* libxml2 */,
				"76E10F9E-9E7B-46AB-ACE1-9D805FD72B07" /* svgtiny */,
			);
			name = libs;
			sourceTree = SOURCE_ROOT;
		};
		BFE7C3452C13750400EE5EDA /* Frameworks */ = {
			isa = PBXGroup;
			children = (
				BFE7C34E2C13751100EE5EDA /* libusb.xcframework */,
				BFE7C34C2C13750E00EE5EDA /* opencv.xcframework */,
				BFE7C34A2C13750B00EE5EDA /* libxml2.xcframework */,
				BFE7C3482C13750800EE5EDA /* assimp.xcframework */,
				BFE7C3462C13750500EE5EDA /* svgtiny.xcframework */,
			);
			name = Frameworks;
			sourceTree = "<group>";
		};
```
```
BFE7C3472C13750500EE5EDA /* svgtiny.xcframework in Frameworks */ = {isa = PBXBuildFile; fileRef = BFE7C3462C13750500EE5EDA /* svgtiny.xcframework */; };
```

VS not working:
```
"044FEE66-08B1-4F5B-B02E-B8C3B28DEF06": {
			"fileEncoding": "4",
			"isa": "PBXFileReference",
			"lastKnownFileType": "folder",
			"path": "../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework",
			"sourceTree": "SOURCE_ROOT"
		},
```

**Comment by dimitre** [comment]
Great. seems easy to fix.

**Comment by dimitre** [comment]
ongoing PR here: 
https://github.com/openframeworks/projectGenerator/pull/520/files

**Comment by danoli3** [comment]
So close now! 

```
"EFF6E6A3-4D2A-445C-B012-92B728EDEC8D" /* assimp.xcframework */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = wrapper.framework; path = assimp.xcframework; sourceTree = "<group>"; };
```

Should be 
```
"EFF6E6A3-4D2A-445C-B012-92B728EDEC8D" /* assimp.xcframework */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = wrapper.xcframework; path = assimp.xcframework; sourceTree = "<group>"; };
```
wrapper.**xc**framework

**Comment by danoli3** [comment]
I'll add to your PR one sec

**Comment by dimitre** [comment]
great. I left you some comments on the code, look for dan there. feel free to update and merge if you find it works

**Comment by danoli3** [comment]
Doing now! Thanks!! Yeah some definite code issues there haha

**Comment by danoli3** [comment]
Fixed for v46 and via https://github.com/openframeworks/projectGenerator/pull/522

**Comment by danoli3** [comment]
> Yeah still issue for xcframework for addonsExample: This is what PG makes atm:
> 
> ```
> "3E4488B0-56D0-472C-B8FA-1AB8716248B5": {
> 			"fileEncoding": "4",
> 			"isa": "PBXFileReference",
> 			"lastKnownFileType": "folder",
> 			"path": "../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework",
> 			"sourceTree": "SOURCE_ROOT"
> 		},
> ```
> 
> Where manually added:
> 
> ```
> 		BF879F872C118EB400200951 /* svgtiny.xcframework */ = {isa = PBXFileReference; lastKnownFileType = wrapper.xcframework; name = svgtiny.xcframework; path = ../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework; sourceTree = "<group>"; };
> ```
> 
> And the build phase changes to:
> 
> ```
> /* Begin PBXFrameworksBuildPhase section */
> 		E4B69B590A3A1756003C02F2 /* Frameworks */ = {
> 			isa = PBXFrameworksBuildPhase;
> 			buildActionMask = 2147483647;
> 			files = (
> 				BF879F7D2C118D6900200951 /* ../../../addons/ofxAssimpModelLoader/libs/assimp/lib/osx/assimp.xcframework in Frameworks */,
> 				BF879F7E2C118D6900200951 /* ../../../addons/ofxOpenCv/libs/opencv/lib/osx/opencv.xcframework in Frameworks */,
> 				BF879F882C118EB400200951 /* svgtiny.xcframework in Frameworks */,
> 				BF879F7F2C118D6900200951 /* ../../../addons/ofxSvg/libs/libxml2/lib/osx/libxml2.xcframework in Frameworks */,
> 			);
> 			runOnlyForDeploymentPostprocessing = 0;
> 		};
> /* End PBXFrameworksBuildPhase section */
> /* Begin PBXGroup section */
> 		"007FA649-C8DE-4E68-ACE2-CD1062614ADF" /* osx */ = {
> 			isa = PBXGroup;
> 			children = (
> 				"3E4488B0-56D0-472C-B8FA-1AB8716248B5" /* ../../../addons/ofxSvg/libs/svgtiny/lib/osx/svgtiny.xcframework */,
> 			);
> 			name = osx;
> 			sourceTree = SOURCE_ROOT;
> 		};
> ```
> 
> Looks like key is: lastKnownFileType = wrapper.xcframework

@dimitre   lastKnownFileType = wrapper.xcframework
